tools/toolchain/Dockerfile

```diff
@@ -13,5 +13,8 @@ RUN dnf -y update \
     && cp system-auth /etc/pam.d \
     && echo 'Defaults !requiretty' >> /etc/sudoers
 RUN mkdir -p /root/.m2/repository
+# we need at least 1.29.164 to run all alternator tests,
+# once on >= fedora 39 we can drop this as it has higer botocore by default
+RUN pip3 install --only-binary=:all: -v --upgrade botocore==1.29.164
```

Comment:
What does "--only-binary" do?
What if we forget to remove this line in the future - will it be ignored if we already have higher botocore, or will it force this specific version?
Finally, I think our usual process isn't to edit this Dockerfile, but rather to edit `install_dependencies.sh`, and that contains the pip and other commands - so I think that's what you should do here. But I want to ask @avikivity to review this part of the patch.

I also want to comment that this change isn't strictly necessary: If we don't do it, it only means that the new test will be skipped. The test won't be skipped for users who manually installed a new botocore - so **you** will be able to test it, and hopefully we won't have a regession before Fedora 39 comes in with an updated botocore and this test returns to the regression testing regimen.

Reply:
--only-binary should install wheel which is recommended way, I guess practical effect is that it will save some space.

--upgrade will force specific version, which should be fine

Putting it in install_dependencies.sh is a bit contradictory with previous requirement that test should use whatever library user has, as people are running this script frequently, but I can move it there if there is such request.

In general I'd like to have this test enabled on CI, as otherwise it could only cause troubles later (i.e. it would be run for the first time on CI when we switch to fedora 39).

I am guessing there is something additional to be done to upgrade image in docker hub after the patch is merged to make the Dockerfile change effective.

Reply:
moved to install_dependencies.sh, also added >= requirement I hope should not downgrade anymore (I am still more in favor of pinning exact version, even exact source of dependencies but that's different discussion)

Reply:
Please ask Fedora to update its botocore version, it's better not to carry too many special dependencies.

Reply:
Exact version is problematic, if people move between branches it forces them to uninstall and reinstall.

There's always the frozen toolchain, but we like not to force it when we can.

Reply:
According to https://packages.fedoraproject.org/pkgs/python-botocore/python3-botocore/
Fedora 38 has 1.29.111, which supposedly is not new enough, and Fedora 39 will have 1.31.21.
I don't know how receptive they are to upgrading Python packages on Fedora 38, you can try.

But personally, I would do what I said before: Just make do with whichever botocore is installed. If it means that your new test will be skipped in Jenkins for the next 6 months, I think it's a small price to pay (this is why I asked to skip the test when botocore isn't new enough to perform it). You tested the new feature yourself on a newer botocore and it passed - it's unlikely that such a small feature will regress in 6 months, and moreover developers with newer botocore (like me) will test it sporadically. When we adopt Fedora 39 or Fedora 38 decides to upgrade botocore, the test will start running automatically.

Reply:
ok, I will remove upgrade of botocore from install-dependecies